server/src/modules/repo/users-repo.service.ts

```diff
@@ -306,6 +306,20 @@ export class UsersRepoService {
         });
     }
 
+    async deleteMapLibraryEntry(userID: number, mapID: number) {
+        const map = await this.prisma.mapLibraryEntry.findFirst({
+            where: {
+                userID: userID,
+                mapID: mapID
+            }
+        });
+        await this.prisma.mapLibraryEntry.delete({
+            where: {
+                id: map.id
+            }
+        });
```

Comment:
Finding the map, *then* deleting it isn't ideal here, we should be able to do it in one go. The reason we can't do it currently is Prisma only lets you run a deletion on a *single* object, i.e. if it's guaranteed an entry is unique. We *could* use a `deleteMany` but that's ugly, the best practice is to write the uniqueness into the schema (MapNotify does this already btw).

So, in `server/prisma/schema.prisma` lets add the line 
```prisma
@@unique([mapID, userID])
```
to the model, below the `@@index`.

Then, we regenerate the prisma client with `npm run migrate` (if it errors, just `rm -rf prisma/migrations/`), and this should be allowed by the new prisma typings:
```ts
await this.prisma.mapLibraryEntry.delete({
    where: {
        mapID_userID: {
            userID: userID,
            mapID: mapID
        }
    }
});
```